scripts/r2s.py

```diff
@@ -112,38 +116,38 @@ def step1():
     tally_num = config.getint('step1', 'tally_num')
     flux_tag = config.get('step1', 'flux_tag')
     decay_times = config.get('step2', 'decay_times').split(',')
-
-    if structured:
-        meshtal = Meshtal(meshtal,
-                        {tally_num: (flux_tag, flux_tag + '_err',
-                                     flux_tag + '_total',
-                                     flux_tag + '_err_total')},
-                        meshes_have_mats=False)
     geom = config.get('step1', 'geom')
     reverse = config.getboolean('step1', 'reverse')
     num_rays = config.getint('step1', 'num_rays')
     grid = config.getboolean('step1', 'grid')
 
     load(geom)
+
+    # get meshtal info from meshtal file
+    flux_mesh = resolve_mesh(meshtal, tally_num, flux_tag)
+
+    # create the cell_fracs array before irradiation_steup
+    if flux_mesh.structured:
+        cell_fracs = discretize_geom(m, num_rays=num_rays, grid=grid)
+        # tag cell fracs for both default and subvoxel r2s modes
+        flux_mesh.tag_cell_fracs(cell_fracs)
+    else:
+        cell_fracs = discretize_geom(flux_mesh)
+
     cell_mats = cell_materials(geom)
-    irradiation_setup(meshtal, cell_mats, alara_params_filename, tally_num,
+    irradiation_setup(flux_mesh, cell_mats, cell_fracs, alara_params_filename, tally_num,
                       num_rays=num_rays, grid=grid, reverse=reverse,
                       flux_tag=flux_tag, decay_times=decay_times,
                       sub_voxel=sub_voxel)
 
     # create a blank mesh for step 2:
-    if structured:
-        mesh = meshtal.tally[tally_num]
-    else:
-        mesh = Mesh(structured=False, mesh=meshtal)
-
-    ves = list(mesh.iter_ve())
+    ves = list(flux_mesh.iter_ve())
     tags_keep = ("cell_number", "cell_fracs",
                  "cell_largest_frac_number", "cell_largest_frac")
-    for tag in mesh.mesh.getAllTags(ves[0]):
+    for tag in flux_mesh.get_all_tags():
         if tag.name not in tags_keep:
-            mesh.mesh.destroyTag(tag, True)
-    mesh.mesh.save('blank_mesh.h5m')
+	     tag.delete()
```

Comment:
An error occurs at this line when running unstructured r2s step1. But I havn't figured out the solution. I put the error message and the example files in https://github.com/CNERG/pyne/pull/51.
The error doesn't happen when running voxel/sub-voxel r2s.

Reply:
This error is fixed in #68 . Test functions could see #51 .